lib/pmdtester/parsers/options.rb

```diff
@@ -0,0 +1,55 @@
+require 'slop'
+
+module PmdTester
+  class Options
+
+    VERSION = '1.0.0'
+
+    attr_reader :local_git_repo
+    attr_reader :base_branch
+    attr_reader :patch_branch
+    attr_reader :base_config
+    attr_reader :patch_config
+    attr_reader :config
+    attr_reader :project_list
+    attr_reader :mode
+
+    def initialize(argv)
+      options = parse(argv)
+      @local_git_repo = options[:r]
+      @base_branch = options[:b]
+      @patch_branch = options[:p]
+      @base_config = options[:bc]
+      @patch_config = options[:pc]
+      @config = options[:c]
+      @project_list = options[:l]
+      @mode = options[:m]
+    end
+
+    private
+
+    def parse(argv)
+      Slop.parse argv do |o|
+        o.string '-r', '--local-git-repo', 'path to the local PMD repository'
```

Comment:
We'll need a more detailed documentation about the options later on. I assume, that you use "-r", if you don't want to clone the PMD repo but rather reuse an existing local clone and thus avoid unnecessary internet traffic.

Reply:
> I assume, that you use "-r", if you don't want to clone the PMD repo but rather reuse an existing local clone and thus avoid unnecessary internet traffic.

Anyways, even if you're going to clone the PMD repo, the address needs to be provided.
Developers are sure to have PMD repo locally, and when we integrate to CI, the clone task should be done with other scripts rather than this tool IMO:)

Reply:
You're right - we always need to provide the repo in some ways.

It seems like Slop also supports marking options as required via `required: true`. Maybe you can mark the mandatory commandline options that way in one of the next PRs?